install_config/upgrading/known_issues.adoc

```diff
@@ -0,0 +1,227 @@
+            "executable": null,
+            "removes": null,
+            "warn": true
+        }
+    },
+    "rc": 1,
+    "start": "2017-08-17 12:45:39.573869"
+}
+
+STDOUT:
+
+error:     rolebindingrestrictions/restriction1 -n namespace1: namespaces "namespace1" not found
+summary: total=1001 errors=1 ignored=0 unchanged=1000 migrated=0
+info: to rerun only failing resources, add --include=rolebindingrestrictions
+error: 1 resources failed to migrate
+----
+
+For this error, a script is provided to automate the following:
+
+- Get all `rolebindingrestriction` objects across all namespaces.
+- Check to see if the associated namespace exists.
+- If the namespace does not exist, temporarily create it.
+- Delete the `rolebindingrestriction` object if the namespace did not exist.
+- Delete the temporary namespace.
+
+To resolve this issue:
+
+. Save the following script to a file, for example *_rolebindingrestrictions.sh_*:
++
+[source,shell]
+----
+IFS=$'\n'
+for rbline in $(oc get rolebindingrestrictions --all-namespaces | tail -n +2); do
+	ns=$(echo "$rbline" | awk '{ printf $1; }')
+	name=$(echo "$rbline" | awk '{ printf $2; }')
+	if ! oc get namespace "$ns"; then
+		echo "Bad namespace: $ns"
+		(oc create ns "$ns"; sleep 10; oc delete rolebindingrestrictions "$name" -n "$ns"; sleep 10; oc delete namespace "$ns") &
+	fi
+done
+unset IFS
+----
+
+. Run the script while logged in to the CLI as a cluster administrator:
++
+----
+$ sh rolebindingrestrictions.sh
+----
+
+[[upgrading-known-issue-BZ1482978]]
+== Orphaned OAuthClientAuthorization Objects
+
+Orphaned `OAuthClientAuthorization` objects can occur during the upgrade due to
+the deletion of the associated OAuth client.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1482978[*BZ#1482978*])
+
+.Error Output
+----
+TASK [Upgrade all storage] *********************************************************************************************
+task path: /usr/share/ansible/openshift-ansible/playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml:11
+Using module file /usr/lib/python2.7/site-packages/ansible/modules/commands/command.py
+
+fatal: [test.linux.lan]: FAILED! => {
+    "changed": true,
+    "cmd": [
+        "oc",
+        "adm",
+        "--config=/etc/origin/master/admin.kubeconfig",
+        "migrate",
+        "storage",
+        "--include=*",
+        "--confirm"
+    ],
+    "delta": "0:00:56.843544",
+    "end": "2017-08-17 12:46:36.417413",
+    "failed": true,
+    "failed_when_result": true,
+    "invocation": {
+        "module_args": {
+            "_raw_params": "oc adm --config=/etc/origin/master/admin.kubeconfig migrate storage --include=* --confirm",
+            "_uses_shell": false,
+            "chdir": null,
+            "creates": null,
+            "executable": null,
+            "removes": null,
+            "warn": true
+        }
+    },
+    "rc": 1,
+    "start": "2017-08-17 12:45:39.573869"
+}
+
+STDOUT:
+
+
+error: <1>
```

Comment:
This is never on separate lines, but I guess this is an OK compromise to have the numbers.

Reply:
Yeah :\